### DISCUSSION TITLE
Tasks stuck in queued state

### DISCUSSION THREAD

**Discussion by NXDnixiaodong** [apache/airflow#31933] [General]
### Apache Airflow version

2.6.1

### What happened

Why are most tasks queued？And even if some running tasks are displayed in the queued state in the k8s cluster

### What you think should happen instead

I think there should be about a hundred tasks running at the same time, but according to my observation, there are only a maximum of forty tasks, and in most cases, only 20 to 30 tasks are running

### How to reproduce

Here are some of my configurations：
```bash
airflow 2.5.1
k8s 1.18.0
executor: KubernetesExecutor 
```
The k8s cluster has one master node and three node nodes

### Operating System

master:Centos7.9/node:Ubuntu20.04

### Deployment

Official Apache Airflow Helm Chart

### Anything else

Simply building a K8S cluster and deploying Airflow should be able to replicate without requiring special operations

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by acmh** [comment]
Since it seems the issue can be consistently replicated, have you tried to reduce an localize the source of the issue? Or could you please provide more input (e.g logs, airflow conf)?

**Reply by NXDnixiaodong** [reply]
Please refer to the configuration below, I plan to upgrade the airflow version to 2.6.1, and I will report back the results after the upgrade, thank you！
```bash
[api]
auth_backends = airflow.api.auth.backend.basic_auth

[celery]
flower_url_prefix = /
worker_concurrency = 16

[celery_kubernetes_executor]
kubernetes_queue = kubernetes

[core]
colored_console_log = False
dags_folder = /opt/airflow/dags
executor = KubernetesExecutor
load_examples = False
parallelism = 10000
remote_logging = False

[elasticsearch]
json_format = True
log_id_template = {dag_id}_{task_id}_{execution_date}_{try_number}

[elasticsearch_configs]
max_retries = 3
retry_timeout = True
timeout = 30

[kerberos]
ccache = /var/kerberos-ccache/cache
keytab = /etc/airflow.keytab
principal = airflow@FOO.COM
reinit_frequency = 3600

[kubernetes]
airflow_configmap = airflow-airflow-config
airflow_local_settings_configmap = airflow-airflow-config
multi_namespace_mode = False
namespace = airflow-test
pod_template_file = /opt/airflow/pod_templates/pod_template_file.yaml
worker_container_repository = apache/airflow
worker_container_tag = 2.5.1

[logging]
colored_console_log = False
remote_logging = False

[metrics]
statsd_host = airflow-statsd
statsd_on = True
statsd_port = 9125
statsd_prefix = airflow

[scheduler]
max_callbacks_per_loop = 500
max_dagruns_per_loop_to_schedule = 500
max_dagruns_to_create_per_loop = 50
run_duration = 41460
standalone_dag_processor = False
statsd_host = airflow-statsd
statsd_on = True
statsd_port = 9125
statsd_prefix = airflow

[webserver]
enable_proxy_fix = True

```

**Comment by potiuk** [comment]
You mentioned 2.6.1 (rather than other) but then below you mentioned 2.5.1. Please upgrade to latest 2.6.* (Tomorrow we are likely to release 2.6.2) and see if it is fixed. See this blog post for details:

https://medium.com/apache-airflow/unsticking-airflow-stuck-queued-tasks-are-no-more-in-2-6-0-6f40a1a22835

**Reply by NXDnixiaodong** [reply]
Sorry for the trouble, the correct airflow version is 2.5.1. In addition, the problem I am currently encountering is indeed similar to the situation described in this blog. I will try to upgrade the airflow version, and I will provide the upgraded results later.

**Comment by potiuk** [comment]
Converted to discussion in case more discussion is needed.
